x/scheduler/keeper/msg_server.go

```diff
@@ -15,3 +21,21 @@ func NewMsgServerImpl(keeper types.Keeper) types.MsgServer {
 }
 
 var _ types.MsgServer = msgServer{}
+
+func (ms msgServer) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
+	if ms.authority != req.Authority {
+		return nil, errors.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", ms.authority, req.Authority)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := ms.SetParams(ctx, req.Params); err != nil {
+		return nil, err
+	}
+
+	return &types.MsgUpdateParamsResponse{}, nil
+}
+
+func (ms msgServer) SetParams(ctx context.Context, params types.Params) error {
+	ms.SetParams(ctx, params)
+	return nil
```

Comment:
Does the `0.50` upgrade enforce this?

Reply:
let me check the usage of this function in the scheduler module and update

Reply:
sorry for the delay, just verified that above functions are required and used in module.go https://github.com/palomachain/paloma/blob/469ba6bf4ecc59010d7c25ce35ce2162ff49c8ac/x/scheduler/module.go#L133

Reply:
removed and pushed